src/btree/bt_sync_obsolete.c

```diff
@@ -388,10 +417,11 @@ __checkpoint_cleanup_walk_btree(WT_SESSION_IMPL *session, WT_ITEM *uri)
     WT_BTREE *btree;
     WT_DECL_RET;
     WT_REF *ref;
-    uint32_t flags;
+    uint32_t flags, num_cc_pages;
 
     ref = NULL;
-    flags = WT_READ_NO_EVICT | WT_READ_VISIBLE_ALL;
+    flags = WT_READ_NO_EVICT | WT_READ_VISIBLE_ALL | WT_READ_WONT_NEED;
```

Comment:
Can we add a comment on why we are having those flags set? For future readers, it would be great to have an explanation of why each flag is necessary for the walk.

Reply:
Done.